unity/Assets/Scripts/ColorChanger.cs

```diff
@@ -1,84 +1,287 @@
+                      potMaterials,
+                      statueMaterials,
+                      watchMaterials,
+
+                      armchairMaterials,
+                      bedMaterials,
+                      chairMaterials,
+                      coffeeTableMaterials,
+                      deskMaterials,
+                      diningTableMaterials,
+                      dresserMaterials,
+                      officeChairMaterials,
+                      shelvingUnitMaterials,
+                      sideTableMaterials,
+                      sofaMaterials,
+
+                      fabricMaterials,
+                      glassMaterials,
+                      lightMaterials,
+                      metalMaterials,
+
+                      paperMaterials,
+                      grungeMaterials,
+                      waxMaterials,
+                      soapMaterials,
+
+                      plasticMaterials,
+                      woodMaterials,
+
+                      wallMaterials,
+                      fridgeMaterials,
+                      microwaveMaterials,
+                      toasterMaterials,
+
+                      lettuceMaterials,
+                      lettuceSlicedMaterials,
+
+                      soapBottleMaterials,
+
+                      potatoMaterials,
+                      potatoCookedMaterials,
+                      potatoSlicedMaterials,
+
+                      laptopMaterials,
+                      breadMaterials,
+                      panMaterials,
+                      panDecalMaterials,
+
+                      coffeeMachineMaterials;
+
+    Dictionary<string, Material[]> materials;
+    Dictionary<string, Color[]> origColors;
+    Dictionary<string, Texture[]> origTextures;
+    Dictionary<string, HashSet<Material>> materialGroups;
 
     public void Start() {
-        targetMaterials = alarmClockMaterials.Concat(appleMaterials).Concat(basketballMaterials).Concat(bowlMaterials).Concat(garbageBinMaterials).Concat(houseplantMaterials).Concat(pillowMaterials).Concat(sprayBottleMaterials).ToArray();
-        backgroundMaterials = bedMaterials.Concat(boxMaterials).Concat(cellphoneMaterials).Concat(cupMaterials).Concat(floorLampMaterials).Concat(penPencilMaterials).Concat(plateMaterials).Concat(potMaterials).Concat(statueMaterials).Concat(watchMaterials).ToArray();
-        furnitureMaterials = armchairMaterials.Concat(bedMaterials).Concat(chairMaterials).Concat(coffeeTableMaterials).Concat(deskMaterials).Concat(diningTableMaterials).Concat(dresserMaterials).Concat(officeChairMaterials).Concat(shelvingUnitMaterials).Concat(sideTableMaterials).Concat(sofaMaterials).ToArray();
-        quickMaterials = ceramicMaterials.Concat(fabricMaterials).Concat(glassMaterials).Concat(lightMaterials).Concat(metalMaterials).Concat(miscMaterials).Concat(plasticMaterials).Concat(woodMaterials).ToArray();
+        materials = new Dictionary<string, Material[]> {
+            ["AlarmClock"] = alarmClockMaterials,
```

Comment:
Can all of these strings be changed to SimObjType enum? Instead of `"AlarmClock"` use `SimObjType.AlarmClock`.

Reply:
Probably not well.. There are many materials that are not object types. For instance, `Grunge`, `Paper`, `Plastic`, `Metal`, `Glass` are not sim object types.

Reply:
Yeah, since not every object with a material that may be swapped is a sim object, I think the way Matt has it set currently is the way to go. It is a bit more tedious if we ever need to update this to include new objects or materials, but it shouldn’t be too bad